.github/workflows/UnitTesting.yaml

```diff
@@ -16,13 +16,14 @@ jobs:
       py39: 3.9
       py310: '3.10'
       py311: '3.11'
+      py312: '3.12'
```

Comment:
FYI: Should have worked at the time with `3.12.0-beta - 3.12`.  But it's available now, see https://github.com/actions/python-versions/blob/main/versions-manifest.json

Maybe rebase to trigger a rerun?